### PR TITLE
Ware balance

### LINKED ISSUE #10074: [BUG] Tasks are not being distributed among multiple warehouses
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### What were you playing at the time? Were you able to reproduce it in both settings?

- [X] Single Player
- [ ] Multi Player

### Minecraft Version

1.20.1

### MineColonies Version

main branch

### Structurize Version

1.20.1-1.0.742-RELEASE

### Related Mods and their Versions

_No response_

### Current Behavior

After the game was updated in early June to distribute the work among couriers evenly, I noticed that all tasks were assigned to one of my five warehouses. I attempted to resolve the issue by: destroying and recreating each warehouse, firing all workers, and rehiring them simultaneously. I've also tried changing the createiveresolve flag as well.

### Expected Behavior

I would expect that tasks would be distributed across the warehouses appropriately, ideally in a way that considers the size of the current task list and the distance from the request to be filled.  I was able to crudely resolve the issue by downloading and building the source and simply shuffling the list of resolvers in RequestHandler.java -> assignRequestDefault. This assigned tasks to the warehouses evenly, although it's not a long term solution.

### Reproduction Steps

1. Create a large colony with multiple warehouses
2. Review the tasks assigned to each warehouse

### Logs

N/A

### Anything else?

This is an impressive piece of software.  Thanks for all you do.

### LINKED ISSUE #10075: [BUG] <Problem about building AI>
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest MineColonies Version?

- [X] I am running the latest **beta** version of MineColonies for my Minecraft version.

### Did you check on the Wiki? or ask on Discord?

- [X] I checked the MineColonies Wiki and made sure my issue is not covered there. Or I was sent from discord to open an issue here.

### Minecraft Version

1.20.1

### MineColonies Version

1.20.1-1.1.603-RELEASE

### Structurize Version

1.20.1-1.0.742-RELEASE

### Related Mods and their Versions

_No response_

### Reproduction Steps

Jsut happended during the game when builder work

### Logs

https://gist.github.com/ink1nk/006550ce486391b078de3d4e2aa2dc46

### Anything else?

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #10075
Closes #10074
Closes #

# Changes proposed in this pull request:
- Warehouse tasks balanced by taking queue size into account
- Fix Small npe
-

[ x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

**Comment by uecasm** [comment]
Isn't this going to make it more likely to have a jumble of every item in every warehouse?  Previously you could force particular items to particular warehouses by controlling which one was closest to the hut that produced it.  Since you need enough of an item in a single warehouse to satisfy a request (and each partial stack still consumes a storage slot, so it wastes space to spread items thinly), this can matter.

**Comment by Raycoms** [comment]
> Isn't this going to make it more likely to have a jumble of every item in every warehouse? Previously you could force particular items to particular warehouses by controlling which one was closest to the hut that produced it. Since you need enough of an item in a single warehouse to satisfy a request (and each partial stack still consumes a storage slot, so it wastes space to spread items thinly), this can matter.

Meant to exclude pickups. Done now

File: src/main/java/com/minecolonies/core/colony/requestsystem/resolvers/DeliverymenRequestResolver.java [hunk 1]
```diff
@@ -66,9 +66,15 @@ public boolean hasCouriers(@NotNull final IRequestManager manager)
...
+    public int getSuitabilityMetric(@NotNull final IRequestManager manager, @NotNull final IRequest<? extends R> request)
     {
-        return (int) BlockPosUtil.getDistance(request.getRequester().getLocation().getInDimensionLocation(), getLocation().getInDimensionLocation());
+        final IWareHouse wareHouse = manager.getColony().getBuildingManager().getBuilding(getLocation().getInDimensionLocation(), IWareHouse.class);
+        final int distance = (int) BlockPosUtil.getDistance(request.getRequester().getLocation().getInDimensionLocation(), getLocation().getInDimensionLocation());
+        if (wareHouse == null)
+        {
+            return distance;
+        }
+        return Math.min(distance/10, 1) + wareHouse.getModule(BuildingModules.WAREHOUSE_REQUEST_QUEUE).getMutableRequestList().size();
```
Comment (someaddons) [inline src/main/java/com/minecolonies/core/colony/requestsystem/resolvers/DeliverymenRequestResolver.java]: if you want to set a minimum you need to use Math.max

Comment (Raycoms) [reply src/main/java/com/minecolonies/core/colony/requestsystem/resolvers/DeliverymenRequestResolver.java]: classic, yes

**Review by someaddons — APPROVED** [review]

**Pull request merged**